src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -12,14 +12,52 @@ export class WebAssemblyResourceLoader {
       credentials: 'include',
       cache: networkFetchCacheMode
     });
+
+    // While we can expect an ASP.NET Core hosted application to include the environment, other
+    // hosts may not. Assume 'Production' in that case.
+    const environment = bootConfigResponse.headers.get('Blazor_Environment') || 'Production';
     const bootConfig: BootJsonData = await bootConfigResponse.json();
     const cache = await getCacheToUseIfEnabled(bootConfig);
 
-    return new WebAssemblyResourceLoader(bootConfig, cache);
+    return new WebAssemblyResourceLoader(bootConfig, environment, cache);
   }
 
-  constructor (public readonly bootConfig: BootJsonData, private cacheIfUsed: Cache | null)
-  {
+  constructor (public readonly bootConfig: BootJsonData, public readonly applicationEnvironment: string, private cacheIfUsed: Cache | null) {
+  }
+
+  readConfigFilesAsync(): Array<ConfigResult>{
+    let results = new Array<ConfigResult>();
```

Comment:
```suggestion
    const results = new Array<ConfigResult>();
```

If it was me, I'd write it as `const results: ConfigResult[] = [];` but then I'm pretty JavaScripty. I'll leave it up to you which style you prefer!